packages/compass-explain-plan/src/components/explain-plan-view.spec.tsx

```diff
@@ -0,0 +1,149 @@
+    winningPlan: {
+      stage: 'COLLSCAN',
+      direction: 'forward',
+    },
+    rejectedPlans: [],
+  },
+  executionStats: {
+    executionSuccess: true,
+    nReturned: 1000,
+    executionTimeMillis: 5,
+    totalKeysExamined: 0,
+    totalDocsExamined: 1000,
+    executionStages: {
+      stage: 'COLLSCAN',
+      nReturned: 1000,
+      executionTimeMillisEstimate: 0,
+      works: 1002,
+      advanced: 1000,
+      needTime: 1,
+      needYield: 0,
+      saveState: 1,
+      restoreState: 1,
+      isEOF: 1,
+      direction: 'forward',
+      docsExamined: 1000,
+    },
+    allPlansExecution: [],
+  },
+  ok: 1,
+} as unknown as Stage).serialize();
+
+describe('ExplainPlanView', function () {
+  afterEach(cleanup);
+
+  it('should render explain plan', function () {
+    render(
+      <ExplainPlanView
+        explainPlan={simpleExplain}
+        rawExplainPlan={simpleExplain.originalExplainData}
+      ></ExplainPlanView>
+    );
+    expect(screen.getByText('COLLSCAN')).to.exist;
+    expect(screen.getByText('Query Performance Summary')).to.exist;
+    expect(screen.getByTestId('docsReturned'))
+      .to.have.property('textContent')
+      .match(/1000\xa0documents returned/);
+    expect(screen.getByTestId('docsExamined'))
+      .to.have.property('textContent')
+      .match(/1000\xa0documents examined/);
+    expect(screen.getByTestId('executionTimeMs'))
+      .to.have.property('textContent')
+      .match(/5\xa0ms\xa0execution time/);
+    expect(screen.getByTestId('sortedInMemory'))
+      .to.have.property('textContent')
+      .match(/Is not\xa0sorted in memory/);
+    expect(screen.getByTestId('indexKeysExamined'))
+      .to.have.property('textContent')
+      .match(/0\xa0index keys examined/);
+  });
+
+  it('should render error when there is no explain plan', function () {
+    render(<ExplainPlanView error="No explain plan"></ExplainPlanView>);
+    expect(screen.getByText('No explain plan')).to.exist;
+  });
+
+  it('should render raw explain when explain was not parsed successfully', function () {
+    render(
+      <ExplainPlanView
+        rawExplainPlan={simpleExplain.originalExplainData}
+        error="Can't parse explain plan"
+      ></ExplainPlanView>
+    );
+    expect(
+      screen.getByText(
+        'Visual explain plan is not supported with this query on this collection in this Compass release',
+        { exact: false }
+      )
+    ).to.exist;
+
+    function parentButton(el: Node | null): Node | null {
+      if (!el) return null;
+      if (el.parentNode?.nodeName === 'BUTTON') return el.parentNode;
+      return parentButton(el.parentNode);
+    }
+
+    expect(parentButton(screen.getByText('Visual Tree'))).to.have.property(
+      'disabled'
+    );
+    expect(parentButton(screen.getByText('Raw Output'))).to.have.property(
+      'disabled'
+    );
+
+    const doc = (
+      screen.getByTestId('raw-explain-plan') as unknown as { _cm: any }
+    )._cm.state.sliceDoc();
```

Comment:
I didn't know we could access cm instance like this. Thanks :)

Reply:
Yeah, for reference this happens here specifically for testing purposes as we can't rely on getting the whole text otherwise because of virtual rendering that codemirror does: https://github.com/mongodb-js/compass/blob/129d5d5375a2815408e12ee15e1b6bd566f52600/packages/compass-editor/src/json-editor.tsx#L847-L848